core/src/main/java/org/apache/iceberg/IncrementalFileCleanup.java

```diff
@@ -273,7 +275,7 @@ private Set<String> findFilesToDelete(
       Set<ManifestFile> manifestsToScan,
       Set<ManifestFile> manifestsToRevert,
       Set<Long> validIds,
-      TableMetadata current) {
+      Map<Integer, PartitionSpec> specsById) {
```

Comment:
+1 for fixing this. In general we should avoid passing huge metadata objects around to helper methods that don't need everything!